auth/couchdb_session_authenticator.go

```diff
@@ -116,23 +119,45 @@ func (a *CouchDbSessionAuthenticator) Validate() error {
 
 // Authenticate adds session authentication cookie to a request.
 func (a *CouchDbSessionAuthenticator) Authenticate(request *http.Request) error {
-
 	a.URL = request.URL.Scheme + "://" + request.URL.Host
 	a.header = request.Header
 	a.ctx = request.Context()
 
-	cookie, err := a.getCookie()
-	if err != nil {
-		return err
+	err := a.refreshCookie()
+	return err
+}
+
+// SetClient sets the http client for the authenticator. It injects a cookiejar
+// if supplied client is missing one.
+func (a *CouchDbSessionAuthenticator) SetClient(client *http.Client) error {
+	var err error
+	if client.Jar == nil {
+		client.Jar, err = cookiejar.New(&cookiejar.Options{PublicSuffixList: publicsuffix.List})
 	}
```

Comment:
Well, in base this is happening to base's http client and here this is done to auth's http client.

It's a bit messy, really - we have to keep client setter public, so base could use it, because this is a separate module and we can't make http client into a required field on auth initiator, because of mapenv constructor.

So if we are allowing setting an http client and want things to work we either inject cookiejar transparently, same as we do in base service, or returning an error if it is not set and then remove creation of default http client in authenticator initiator. I prefer to inject to have the logic in-line with base service and honestly, because this is a minimal change for a price of a bit of code duplication.

Another approach could be to move the whole thing into `common` and make client field private, but that's bigger refactoring and a breaking change.

Reply:
This concern is resolved by the newer changes.